### PR TITLE
Issue #845: Core menu blocks

### PR DESCRIPTION
**Pull request by docwilmot**
First review Fix for https://github.com/backdrop/backdrop-issues/issues/845

### CODE REVIEW

File: core/modules/system/system.menu.inc [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+      '6' => '6',
+      '7' => '7',
+      '8' => '8',
+      '9' => '9',
+      '0' => t('Unlimited'),
+    ),
+    '#description' => t('From the starting level, specify the maximum depth of the menu tree.'),
+  );
+
+  $form['#validate'][] = 'node_search_validatev';
```
Comment (quicksketch) [inline core/modules/system/system.menu.inc]: Looks like this should be removed or fixed? `node_search_validatev` probably isn't the right validation here.

File: core/modules/system/system.menu.inc [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+function system_menu_configure_form($config) {
+  // Get the config from the form state.
+  if (!empty($config)) {
+    if (!empty($config['parent'])) {
+      list($config['menu_name'], $config['parent_mlid']) = explode(':', $config['parent']);
+    }
+  }
+
+  // Build the standard form.
+  //$form['#attached']['js'][] = backdrop_get_path('module', 'system') . '/menu-block.js';
```
Comment (quicksketch) [inline core/modules/system/system.menu.inc]: No longer necessary I don't think.

File: core/modules/system/system.menu.inc [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    'parent' => 'main-menu:0',
+    'title_link' => 0,
+    'admin_title' => '',
+    'level' => 1,
+    'follow' => 0,
+    'depth' => 0,
+    'expanded' => 0,
+    'sort' => 0,
+    'menu_name' => 'main-menu',
+    'parent_mlid' => 0,
```
Comment (quicksketch) [inline core/modules/system/system.menu.inc]: These options can be trimmed down substantially. Now that we're only providing two options, I think `level` and `depth` are the only necessary options?

File: core/modules/system/system.menu.inc [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+ *   - depth: (int) The maximum depth the tree should contain, relative to the
+ *     starting level.
+ *
+ * @return array
+ *   An associative array containing several pieces of data.
+ *   - content: The tree as a renderable array.
+ *   - subject: The title rendered as HTML.
+ *   - subject_array: The title as a renderable array.
+ */
+function system_menu_tree_build(array &$config) {
```
Comment (quicksketch) [inline core/modules/system/system.menu.inc]: Does `$config` need to be by-reference here?

**Comment by quicksketch** [comment]
This is looking great!

**Comment by hellomrcat** [comment]
@docwilmot this just needs an updated commit message like `Issue #845: Add menu_block features to core.` but otherwise looks good!

**Comment by quicksketch** [comment]
I left a bit of new feedback at https://github.com/backdrop/backdrop-issues/issues/845#issuecomment-168595612.

**Comment by quicksketch** [comment]
This was replaced by https://github.com/backdrop/backdrop/pull/1322 and then https://github.com/backdrop/backdrop/pull/1375, which is now merged. Thanks!

**Pull request closed without merging**